humblegen/src/ast.rs

```diff
@@ -106,26 +122,88 @@ impl VariantDef {
     }
 }
 
-/// A field node (field definition inside struct).
 #[derive(Debug)]
+pub struct ServiceDef {
+    pub name: String,
+    pub doc_comment: Option<String>,
+    pub endpoints: Vec<ServiceEndpoint>,
+}
+
+#[derive(Debug)]
+pub struct ServiceEndpoint {
+    pub doc_comment: Option<String>,
+    pub route: ServiceRoute,
+}
+
+#[derive(Debug)]
+pub enum ServiceRoute {
```

Comment:
Do we support `DELETE` by the way? It's probably the third most common verb, followed by `PUT` then `PATCH`, in my intuition.

Reply:
Should a `DELETE` request have a return type?
And how should non-existence of the to-be-deleted resource be represented? As a domain error (like we do for the other endpoints?)?
---

```
    http_delete ~ http_route ~ http_query? ~ "->" ~ type_ident
```
? 

Or 
```
    http_delete ~ http_route ~ http_query?
```

Reply:
> Should a DELETE request have a return type?

This is the same as for `POST`, possible, but not required.

Did we give this enough thought by the way? It's common (but not required) to return a 204 in these situations, which would also be valid for a `POST`. Maybe special case it for return values of `option[T]`, making `None` return an empty response?

> And how should non-existence of the to-be-deleted resource be represented? As a domain error (like we do for the other endpoints?)?

It's a domain error, since it is entirely reasonable to have both, APIs that return an error when trying to delete a non-existant entity as well as APIs that silently discard the ineffective deletion.